### PR TITLE
Re Add ReviewBackgroundHostedService.cs

### PR DESCRIPTION
**Pull request by chidozieononiwu**
This add `ReviewBackgroundHostedService` for updating reviews now apiRevisoins.

Depends on merge of https://github.com/Azure/azure-sdk-tools/pull/7303/files#diff-b0398569b9b71cf1282bf6e43715139075ab4d1adfe46b7b512434b7097858ab

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs [hunk 1]
```diff
@@ -425,6 +427,88 @@ public async Task<bool> IsAPIRevisionTheSame(APIRevisionListItemModel revision,
...
+        }
+        /// <summary>
+        /// SoftDelete APIRevision if its not been updated after many months
+        /// </summary>
+        /// <param name="archiveAfterMonths"></param>
+        /// <returns></returns>
+        public async Task AutoArchiveAPIRevisions(int archiveAfterMonths)
+        {
+            var lastUpdatedDate = DateTime.Now.Subtract(TimeSpan.FromDays(archiveAfterMonths * 30));
+            var manualRevisions = await _apiRevisionsRepository.GetAPIRevisionsAsync(lastUpdatedOn: lastUpdatedDate, apiRevisionType:  APIRevisionType.Manual);
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs:475]: We should archive pull request reviews here. Not manual reviews.

File: src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs [hunk 2]
```diff
@@ -268,6 +272,66 @@ public async Task<int> GenerateAIReview(string reviewId, string revisionId)
...
+            foreach (var language in LanguageService.SupportedLanguages)
+            {
+                if (updateDisabledLanguages.Contains(language))
+                {
+                    _telemetryClient.TrackTrace("Background task to update API review at startup is disabled for langauge " + language);
+                    continue;
+                }
+                var languageService = LanguageServiceHelpers.GetLanguageService(language, _languageServices);
+                if (languageService == null)
+                    return;
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs:292]:
```suggestion
                    continue;
```

File: src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs (hunk 1 above)
Comment (chidozieononiwu) [reply src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs:475]: Are you sure about this. It used to be Manual reviews https://github.com/Azure/azure-sdk-tools/blob/8b4943ac480d9cd3c699507d8cfb1221cd9c8a01/src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs#L509

File: src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs (hunk 2 above)
Comment (chidozieononiwu) [reply src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs:292]: Sure about this? https://github.com/Azure/azure-sdk-tools/blob/8b4943ac480d9cd3c699507d8cfb1221cd9c8a01/src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs#L361

Comment (praveenkuttappan) [reply src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs:292]: Yes, It's a bug in current code if we are returning.

**Pull request merged**